src/unix/tcp.c

```diff
@@ -153,7 +153,7 @@ int uv__tcp_connect(uv_connect_t* req,
   assert(handle->type == UV_TCP);
 
   if (handle->connect_req != NULL)
-    return -EALREADY;  /* FIXME(bnoordhuis) -EINVAL or maybe -EBUSY. */
+    return -UV_EBUSY;
```

Comment:
You should return -EBUSY here.  Here is how we do errors in libuv:

* on unix a UV error is the negated errno, that is, UV_EBUSY == -EBUSY, and an error will always be < 0
* on windows we define a set of negative constants so an error is also always < 0